doc/user/content/sql/tail.md

```diff
@@ -66,33 +136,67 @@ Syntax has changed. `WITH SNAPSHOT` is now `WITH (SNAPSHOT)`.
+  - If the tailed relation is materialized, pick the latest time for which
+    results are computed.
+  - If the tailed relation is unmaterialized, pick time `0`.
 
-### SNAPSHOT
+All involved sources must be valid to read from at the chosen timestamp. A given
+timestamp will be rejected if data it would report has already been compacted by
+Materialize. See the [`--logical-compaction-window`](/cli/#compaction-window)
+command-line option for details on Materialize's compaction policy.
 
-By default, each TAIL is created with a `SNAPSHOT` which contains the results of the query at its `AS OF` timestamp.
-Any further updates to these results are produced at the time when they occur. To only see results after the
-`AS OF` timestamp, specify `WITH (SNAPSHOT = false)`.
+### `SNAPSHOT`
 
-### PROGRESS
+By default, each `TAIL` begins by emitting a snapshot of the tailed relation,
+which contains the contents of the relation at its `AS OF` timestamp. Any further
+updates to these results are produced at the time when they occur.
 
-If the `PROGRESS` option is specified (`WITH (PROGRESS)`) an additional `progressed` column appears in the output.
+To only see results after the [`AS OF`](#as-of) timestamp, specify `WITH
+(SNAPSHOT = false)`.
+
+### `PROGRESS`
+
+If the `PROGRESS` option is specified via `WITH (PROGRESS)`, an additional
+`progressed` column appears in the output.
 It is `false` if there may be more rows with the same timestamp.
-It is `true` if no more timestamps will appear less than the timestamp.
+It is `true` if no more timestamps will appear that are strictly less than the
+timestamp.
 All further columns after `progressed` will be `NULL` in the `true` case.
 
-Not all timestamps that appear will have a corresponding `done` message.
-For example timestamps `1`, `2`, and `3` may appear with only a single `done` message for `3`.
+Intuitively, progress messages are used to explicitly communicate that no
+updates have occurred in a given time window. Without explicit progress
+messages, it is impossible to distinguish between a stall in Materialize and
+a legimate period of no updates.
+
+Not all timestamps that appear will have a corresponding `progressed` row.
+For example, the following is a valid sequence of updates:
+
+`timestamp` | `progressed` | `diff` | `column1`
+------------|--------------|--------|----------------
+1           | `false`      | 1      | data
+1           | `false`      | 1      | more data
+2           | `false`      | 1      | even more data
+4           | `true`       | `NULL` | `NULL`
+
+Notice how Materialize did not emit explicit progress messages for timestamps
+`1`, `2`, or `3`. The receipt of the update at timestamp `2` implies that there
+are no more updates for timestamp `1`, because timestamps are always presented
+in non-decreasing order. The receipt of the explicit progress message at
+timestamp `4` implies that there are no more updates for either timestamp
+`2` or `3`.
```

Comment:
I'd belabor the point here and state that you might expect to see messages with a timestamp of 4.